crates/snark-acct-types/src/outputs.rs

```diff
@@ -36,6 +36,20 @@ impl UpdateOutputs {
     pub fn messages_mut(&mut self) -> &mut Vec<OutputMessage> {
         &mut self.messages
     }
+
+    pub fn total_output_value(&self) -> Option<BitcoinAmount> {
+        let mut total_sent = BitcoinAmount::zero();
+
+        for t in self.transfers() {
+            total_sent = total_sent.checked_add(t.value())?;
+        }
+
+        for m in self.messages() {
+            total_sent = total_sent.checked_add(m.payload().value())?;
+        }
+
+        Some(total_sent)
+    }
```

Comment:
Should be a verby function name (`compute_` or at least `get_`) since it's doing something nontrivial.